src/bun.js/bindings/CPUFeatures.cpp

```diff
@@ -68,48 +160,66 @@ static uint8_t x86_cpu_features()
+#if OS(DARWIN)
+    // On macOS/iOS, use sysctlbyname to detect CPU features
     int value = 0;
     size_t size = sizeof(value);
-    if (sysctlbyname("hw.optional.AdvSIMD", &value, &size, NULL, 0) == 0 && value == 1)
-        features |= 1 << static_cast<uint8_t>(AArch64CPUFeature::neon);
-    if (sysctlbyname("hw.optional.floatingpoint", &value, &size, NULL, 0) == 0 && value == 1)
-        features |= 1 << static_cast<uint8_t>(AArch64CPUFeature::fp);
-    if (sysctlbyname("hw.optional.arm.FEAT_AES", &value, &size, NULL, 0) == 0 && value == 1)
+
+    // Check for AES
+    if (sysctlbyname("hw.optional.arm.FEAT_AES", &value, &size, nullptr, 0) == 0 && value == 1)
         features |= 1 << static_cast<uint8_t>(AArch64CPUFeature::aes);
-    if (sysctlbyname("hw.optional.armv8_crc32", &value, &size, NULL, 0) == 0 && value == 1)
+
+    // Check for CRC32
+    if (sysctlbyname("hw.optional.arm.FEAT_CRC32", &value, &size, nullptr, 0) == 0 && value == 1)
         features |= 1 << static_cast<uint8_t>(AArch64CPUFeature::crc32);
-    if (sysctlbyname("hw.optional.arm.FEAT_LSE", &value, &size, NULL, 0) == 0 && value == 1)
+
+    // Check for LSE/Atomics
+    if (sysctlbyname("hw.optional.arm.FEAT_LSE", &value, &size, nullptr, 0) == 0 && value == 1)
         features |= 1 << static_cast<uint8_t>(AArch64CPUFeature::atomics);
-    if (sysctlbyname("hw.optional.arm.FEAT_SVE", &value, &size, NULL, 0) == 0 && value == 1)
+
+    // Check for SVE
+    if (sysctlbyname("hw.optional.arm.FEAT_SVE", &value, &size, nullptr, 0) == 0 && value == 1)
         features |= 1 << static_cast<uint8_t>(AArch64CPUFeature::sve);
-#elif OS(LINUX)
-    unsigned long hwcaps = getauxval(AT_HWCAP);
-    if (hwcaps & HWCAP_ASIMD)
-        features |= 1 << static_cast<uint8_t>(AArch64CPUFeature::neon);
-    if (hwcaps & HWCAP_FP)
-        features |= 1 << static_cast<uint8_t>(AArch64CPUFeature::fp);
-    if (hwcaps & HWCAP_AES)
+#else
+    // For non-Apple ARM64 platforms, we can use the system register approach
+    // but we need to be careful about illegal instructions
+
+    uint64_t id_aa64isar0_el1 = 0;
+    uint64_t id_aa64isar1_el1 = 0;
+    uint64_t id_aa64pfr0_el1 = 0;
+
+    // Use inline assembly with constraints to safely read system registers
+    asm volatile("mrs %0, id_aa64isar0_el1" : "=r"(id_aa64isar0_el1));
+    asm volatile("mrs %0, id_aa64isar1_el1" : "=r"(id_aa64isar1_el1));
+    asm volatile("mrs %0, id_aa64pfr0_el1" : "=r"(id_aa64pfr0_el1));
+
+    // Check for AES (bits 7:4 of ID_AA64ISAR0_EL1)
+    if (((id_aa64isar0_el1 >> 4) & 0xf) > 0)
         features |= 1 << static_cast<uint8_t>(AArch64CPUFeature::aes);
-    if (hwcaps & HWCAP_CRC32)
+
+    // Check for CRC32 (bits 19:16 of ID_AA64ISAR0_EL1)
+    if (((id_aa64isar0_el1 >> 16) & 0xf) > 0)
         features |= 1 << static_cast<uint8_t>(AArch64CPUFeature::crc32);
-    if (hwcaps & HWCAP_ATOMICS)
+
+    // Check for LSE/Atomics (bits 23:20 of ID_AA64ISAR0_EL1)
+    if (((id_aa64isar0_el1 >> 20) & 0xf) > 0)
```

Comment:
```suggestion
    // Check for LSE/Atomics (bits 23:20 of ID_AA64ISAR0_EL1 are 0b0010 or higher)
    if (((id_aa64isar0_el1 >> 20) & 0xf) >= 2)
```

per [docs](https://developer.arm.com/documentation/ddi0601/2024-12/AArch64-Registers/ID-AA64ISAR0-EL1--AArch64-Instruction-Set-Attribute-Register-0), 0b0010 and higher indicate FEAT_LSE, 0b0001 is not defined